mobile/src/components/courses/courses.tsx

```diff
@@ -33,8 +33,10 @@ const Courses: FC = (): ReactElement => {
     dispatch(coursesActions.getCourses({ title: '', categoryKey: '' }));
   };
 
-  const handleCourseCard = (id: CourseGetRequestParamsDto): void => {
-    dispatch(coursesActions.getCourse(id));
+  const handleCourseCard = async (
+    id: CourseGetRequestParamsDto,
+  ): Promise<void> => {
+    await dispatch(coursesActions.getCourse(id)).unwrap();
```

Comment:
It's better to navigate to this course and add loader in it while course data is loading